File: __tests__/utils/orientationUtils.test.ts
import {ORIENTATION_LABEL, ORIENTATION_FR} from '../../src/utils/orientationUtils';
import {Orientation} from '../../src/types/sector';

const ALL_ORIENTATIONS: Orientation[] = ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW'];

describe('ORIENTATION_LABEL', () => {
  it.each(ALL_ORIENTATIONS)('orientation %s a une entrée', o => {
    expect(ORIENTATION_LABEL[o]).toBeDefined();
  });

  it.each(ALL_ORIENTATIONS)('le label %s contient la lettre directionnelle', o => {
    expect(ORIENTATION_LABEL[o]).toContain(o);
  });

  it.each<[Orientation, string]>([
    ['N',  '↑'],
    ['NE', '↗'],
    ['E',  '→'],
    ['SE', '↘'],
    ['S',  '↓'],
    ['SW', '↙'],
    ['W',  '←'],
    ['NW', '↖'],
  ])('ORIENTATION_LABEL[%s] contient la flèche %s', (o, arrow) => {
    expect(ORIENTATION_LABEL[o]).toContain(arrow);
  });
});

describe('ORIENTATION_FR', () => {
  it.each(ALL_ORIENTATIONS)('orientation %s a un nom français', o => {
    expect(ORIENTATION_FR[o]).toBeDefined();
  });

  it.each<[Orientation, string]>([
    ['N',  'Nord'],
    ['NE', 'Nord-Est'],
    ['E',  'Est'],
    ['SE', 'Sud-Est'],
    ['S',  'Sud'],
    ['SW', 'Sud-Ouest'],
    ['W',  'Ouest'],
    ['NW', 'Nord-Ouest'],
  ])('ORIENTATION_FR[%s] === "%s"', (o, expected) => {
    expect(ORIENTATION_FR[o]).toBe(expected);
  });
});

File: app/src/test/java/com/almostblue/domain/OrientationUtilsTest.kt
package com.almostblue.domain

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

/** Port 1:1 de spec/__tests__/utils/orientationUtils.test.ts. */
class OrientationUtilsTest {

    @Test
    fun `chaque label contient la lettre directionnelle`() {
        for (o in Orientation.entries) {
            assertTrue("label de $o", o.label.contains(o.name))
        }
    }

    @Test
    fun `chaque label contient la bonne fleche`() {
        val arrows = mapOf(
            Orientation.N to "↑", Orientation.NE to "↗", Orientation.E to "→",
            Orientation.SE to "↘", Orientation.S to "↓", Orientation.SW to "↙",
            Orientation.W to "←", Orientation.NW to "↖",
        )
        for ((o, arrow) in arrows) {
            assertTrue("flèche de $o", o.label.contains(arrow))
        }
    }

    @Test
    fun `noms francais exacts`() {
        val expected = mapOf(
            Orientation.N to "Nord", Orientation.NE to "Nord-Est", Orientation.E to "Est",
            Orientation.SE to "Sud-Est", Orientation.S to "Sud", Orientation.SW to "Sud-Ouest",
            Orientation.W to "Ouest", Orientation.NW to "Nord-Ouest",
        )
        for ((o, name) in expected) {
            assertEquals(name, o.frenchName)
        }
    }
}
